src/providers/auth.ts

```diff
@@ -74,33 +75,40 @@ export class FirebaseAuth extends ReplaySubject<FirebaseAuthState> {
 
     switch (config.method) {
       case AuthMethods.Popup:
-        return this._authBackend.authWithOAuthPopup(config.provider, this._scrubConfig(config));
+        return this._authBackend.authWithOAuthPopup(config.provider, this._scrubConfig(config))
+          .then((userCredential: firebase.auth.UserCredential) => {
+            return authDataToAuthState(userCredential.user, <OAuthCredential>(<any>userCredential).credential)
```

Comment:
Need to be consistent in where the call to `authDataToAuthState` happens, either here or in `auth_backend`/`firebase_sdk_backend`.